client/components/FaveButton.tsx

```diff
@@ -19,15 +19,40 @@ const FaveButton = ({ zoop }: FaveButtonProps) => {
   const currentUser = useSelector((state: RootState) => state.auth.user)
 
   const [addFave, {isLoading, data, error}] = useAddFaveMutation();
+  const [
+    removeFave, 
+    {
+      isLoading: isRemoveFaveLoading, 
+      data: removeFaveData, 
+      error: removeFaveError
+    }] = useRemoveFaveMutation();
 
 
+  const currentUserFave = zoop.faves.filter(
+    fave => fave.faverId === currentUser?.id
+  )
+
   const handleClick = ((e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
     e.preventDefault();
-    if (!isLoading && currentUser) {
-      addFave({
-        faverId: currentUser.id,
-        zoopId: zoop.id
-      })
+    if (!isLoading && !isRemoveFaveLoading && currentUser) {
+
+      if (!currentUserFave.length) {
```

Comment:
I'd recommend find or some above and changing this to a truthy-check on currentUserFave